### PR TITLE
Preliminary support for Python 3

### PR DESCRIPTION
**Pull request by viotti**
I did some flashing and formatting on the CC3200 LaunchPad and
everything seems to work. I did not perform an extensive test of the
tool though.

Tested on Python 2.7 and Python 3.6.

### CODE REVIEW

File: cc3200tool/cc.py [hunk 1]
```diff
@@ -434,6 +474,8 @@ def _raw_write(self, offset, data, storage_id=0):
             raise CC3200Error("no serial flash?!")
 
         sinfo = self._get_storage_info()
+
+        # Whats the purpose of "bs", "start" and "count" here?!?
```
Comment (kzyapkov) [inline cc3200tool/cc.py]: yeah, I have no idea. Maybe just drop the block

File: cc3200tool/cc.py [hunk 2]
```diff
@@ -568,8 +610,12 @@ def format_slfs(self, size=None):
         size = SLFS_SIZE_MAP[size]
 
         log.info("Formatting flash with size=%s", size)
-        command = OPCODE_FORMAT_FLASH \
-            + struct.pack(">IIIII", 2, size/4, 0, 0, 2)
+        if PY3:
+            command = OPCODE_FORMAT_FLASH \
+                + struct.pack(">IIIII", 2, size//4, 0, 0, 2)
```
Comment (kzyapkov) [inline cc3200tool/cc.py]: doesn't the `//` operator work equally well in python 2?

**Review by kzyapkov — COMMENTED** [review]
I'm wondering if it's possible to use `bytearray` for data in and out of the serial port, so we can treat it uniformly throughout the code. This should reduce the number of `if PY3` branches.

**Comment by kzyapkov** [comment]
Also, thanks for doing this!

File: cc3200tool/cc.py (hunk 2 above)
Comment (viotti) [reply cc3200tool/cc.py]: Yes, I think `//` works for both Python 2 and 3. I'll remove the `if PY3:` test here.

File: cc3200tool/cc.py (hunk 1 above)
Comment (viotti) [reply cc3200tool/cc.py]: OK, I'll drop this block.

**Comment by viotti** [comment]
You're welcome. I pushed a couple of fixes.

I believe that a `bytearray` would indeed reduce the amount of version checks, if not eliminate them altogether. Indexing a `bytearray` object produces integers in both major versions of the language. 

    $ python
    Python 2.7.10 (v2.7.10:15c95b7d81dc, May 23 2015, 09:33:12)
    [GCC 4.2.1 (Apple Inc. build 5666) (dot 3)] on darwin
    Type "help", "copyright", "credits" or "license" for more information.
    >>> bytearray('BYTEARRAY')
    bytearray(b'BYTEARRAY')
    >>> bytearray('BYTEARRAY')[4]
    65
    >>> ^D
    $ python3
    Python 3.6.0 (v3.6.0:41df79263a11, Dec 22 2016, 17:23:13)
    [GCC 4.2.1 (Apple Inc. build 5666) (dot 3)] on darwin
    Type "help", "copyright", "credits" or "license" for more information.
    >>> bytearray(b'BYTEARRAY')
    bytearray(b'BYTEARRAY')
    >>> bytearray(b'BYTEARRAY')[4]
    65
    >>> ^D

All checks I've introduced arise from the fact that indexing `str` on Python two produces `str`. And the `bytearray` would probably make the `memoryview` uses at lines 202-206 expendable.

**Comment by mon** [comment]
With Python 2 EOL a lot closer than it was in 2017, is it worth resurrecting this PR?

**Comment by kzyapkov** [comment]
Indeed. I'm not using this tool actively anymore, but will try to find a spare minute and get this merged in. Contributions are welcome! I think there are forks which do Python3 already ...

**Comment by kzyapkov** [comment]
We've dropped Py2 entirely, closing this.

**Pull request closed without merging**